### PR TITLE
Allow form fields' `label` to accept any node (#526)

### LINKED ISSUE #526: Allow to pass label as `node` into form elements
We need to add the possibility to pass either `string` or `node` into the `label` prop of form field components.

This is due to a requirement to use the label as a trigger for showing a `Popover` component.

Change needs to take place on:
* https://react-ui.io/components/CheckboxField/
* https://react-ui.io/components/FileInputField/
* https://react-ui.io/components/Radio/
* https://react-ui.io/components/SelectField/
* https://react-ui.io/components/TextArea/
* https://react-ui.io/components/TextField/
* https://react-ui.io/components/Toggle/

### PR DESCRIPTION
**Pull request by bedrich-schindler**
Closes #526

### CODE REVIEW

**Comment by bedrich-schindler** [comment]
Documentation does not specifically mention data type of labels, only in the props table.  @adamkudrna @mbohal @dacerondrej, do you think we need to extend documentation of those components to mention this possibility or not? Personally, I do not see any additional value to this.

I expect to extend documentation of Popover to mention possibility to extend label with Popover (https://github.com/react-ui-org/react-ui/issues/489), but do not know what to say about it in documentation of inputs ...

File: src/components/CheckboxField/CheckboxField.jsx
Comment (mbohal) [inline src/components/CheckboxField/CheckboxField.jsx:1]: I think we also need to add support for the `InputGroup` component.

Comment (mbohal) [inline src/components/CheckboxField/CheckboxField.jsx:1]: I think we should document this with code examples.

**Review by mbohal — CHANGES_REQUESTED** [review]

File: src/components/CheckboxField/CheckboxField.jsx
Comment (mbohal) [reply src/components/CheckboxField/CheckboxField.jsx:1]: I checked and I think the `InputGroup` is the only one, where we want to include it.

Other components mostly render `label` "within" something (e.g. `Badge`, `Button`). Then there is `TextLink` where the name kind of implies text prop.

Comment (bedrich-schindler) [reply src/components/CheckboxField/CheckboxField.jsx:1]: Please check my comment: https://github.com/react-ui-org/react-ui/pull/543#issuecomment-2139112900

Personally, I don't think so we need to show examples of everything we support. End user must know React, so he/she must know what React.Node means and therefore I do not see any reason to have it as example. What more, I think user should use always except situations when he/she needs to provide additional info about the field. I can imagine to extend prop documentation or general guidelines with something like "Use text labels unless it is necessary to wrap text label into Popover-like to  component to provide additional info about the field". 

There was separate issue for that I guess but I think it disappeared from the board after Monday's meeting.

What do you think @mbohal @adamkudrna @dacerondrej

**Review by dacerondrej — APPROVED** [review]

File: src/components/CheckboxField/CheckboxField.jsx
Comment (mbohal) [reply src/components/CheckboxField/CheckboxField.jsx:1]: Ok, lets put it in general guidelines

File: src/components/FileInputField/FileInputField.jsx [hunk 1]
```diff
@@ -123,9 +123,9 @@ FileInputField.propTypes = {
    */
   isLabelVisible: PropTypes.bool,
   /**
-   * Text field label.
+   * File input field label.
```
Comment (adamkudrna) [inline src/components/FileInputField/FileInputField.jsx:126]: Thanks!

**Review by adamkudrna — APPROVED** [review]

**Pull request merged**